### ISSUE TITLE
Installation under MacOS

### ISSUE THREAD

**Issue by HeisenbergK** [eminfedar/async-sockets-cpp#18]
Hello,

I am trying to install under MacOS.

The make command compiles and after I run `sudo ./install.sh`, the header files get copied as they should.

But, the .so file cannot be copied cause the target directory does not exist. Also I think it shouldn't be .so, cause Mac use dylib files.

**Comment by eminfedar** [comment]
I made it header-only (ac946e44fff3808d7556fda94a36a77aadc37b0c)  so no need to use .so libraries.

I don't have macOS, if someone update the `install.sh` file to adapt macOS too, I will gladly merge it.

**Comment by ChristianGabs** [comment]
So let's verify your issues,


Step 1 ) let clone the project

```shell
Developments-iMac:test root# pwd
/Users/cristiang/test
Developments-iMac:test root# git clone https://github.com/eminfedar/async-sockets-cpp.git .
Cloning into '.'...
remote: Enumerating objects: 381, done.
remote: Counting objects: 100% (45/45), done.
remote: Compressing objects: 100% (34/34), done.
remote: Total 381 (delta 26), reused 21 (delta 11), pack-reused 336
Receiving objects: 100% (381/381), 332.62 KiB | 4.82 MiB/s, done.
Resolving deltas: 100% (216/216), done.
```

2) Let install our headers

```shell
Developments-iMac:test root# ls
.git            .gitignore      LICENSE         README.md       async-sockets   examples        img
Developments-iMac:test root# cd async-sockets/
Developments-iMac:async-sockets root# ls
include         install.sh      remove.sh
Developments-iMac:async-sockets root# chmod +x install.sh
Developments-iMac:async-sockets root# ./install.sh
Info: The library has been copied to -> /usr/local/include

Example usage:
#include <async-sockets/tcp-client.hpp>
Developments-iMac:async-sockets root# ls /usr/local/include/async-sockets/
basesocket.hpp  tcpserver.hpp   tcpsocket.hpp   udpserver.hpp   udpsocket.hpp
Developments-iMac:async-sockets root#
```

3) Let Compile one of the example

```shell
Developments-iMac:test root# ls
.git            .gitignore      LICENSE         README.md       async-sockets   examples        img
Developments-iMac:test root# cd examples/
Developments-iMac:examples root# ls
CMakeLists.txt  Makefile        tcp-client.cpp  tcp-server.cpp  udp-client.cpp  udp-server.cpp
Developments-iMac:examples root# make tcp-server
g++ --std=c++11 tcp-server.cpp -I../async-sockets/include -lpthread -o tcp-server
Developments-iMac:examples root# ls
CMakeLists.txt  Makefile        tcp-client.cpp  tcp-server      tcp-server.cpp  udp-client.cpp  udp-server.cpp
Developments-iMac:examples root# ./tcp-server
New client: [172.30.1.21:54006]
172.30.1.21:54006 => hi

172.30.1.21:54006 => ▒▒▒▒
Socket closed:172.30.1.21:54006 -> 0
```

If you provide more info on your steps I will be more happy to help

**Comment by eminfedar** [comment]
Thanks for your clear explanations :) @ChristianGabs
